src/data_info/services/rename.clj

```diff
@@ -22,42 +22,55 @@
+   ;; data-restore: paths, restoration-paths
+   ;; Then we need to check the paths. No source or destination can be the source or destination of another move.
+   ;; This includes exact matches but also prefix matches in both directions:
+   ;; If a locked source plus '/' is a prefix of a new source, we're moving the new source with both processes
+   ;; If a locked destination plus '/' is a prefix of a new source, we're trying to move the new source out while it's still being moved in
+   ;; If a new source plus '/' is a prefix of a locked source, we're moving the locked source with both processes
+   ;; If a new source plus '/' is a prefix of a locked destination, we're  trying to move something in while part of it (the locked dest) is being moved out
+   ;; Locked source + '/' prefix of new dest, we're moving new dest in while the parent is being moved out
+   ;; Locked dest + '/' prefix of new dest, we're moving stuff into new dest with two processes
+   ;; New dest + '/' prefix of locked source, we might be moving stuff into locked source while it's being moved out
+   ;; New dest + '/' prefix of locked dest, we're moving stuff into locked dest with two processes
+   (let [far-future "9999-12-31T23:59:59Z"
+         eligible-async-task-types ["data-move" "data-rename" "data-delete" "data-delete-trash" "data-restore"]
+         eligible-tasks (async-tasks/get-by-filter {:type eligible-async-task-types
+                                                    :include_null_end true
+                                                    :end_date_since far-future})
+         add-destination-to-basenames (fn [destination sources]
+                                        (map #(ft/path-join destination %) (map ft/basename sources)))
+         extract-paths (fn [{:keys [data type]}]
+                         (condp = type
+                           "data-move"         (concat (add-destination-to-basenames (:destination data) (:sources data))
+                                                       (:sources data))
+                           "data-rename"       (map #(get data %) [:destination :source])
+                           "data-delete"       (concat (:paths data) (vals (:trash-paths data)))
+                           "data-delete-trash" (:trash-paths data)
+                           "data-restore"      (concat (:paths data)
+                                                       (map :restored-path (vals (:restoration-paths data))))
+                           nil))
+         locked-paths (reduce conj #{} (mapcat extract-paths eligible-tasks))
+         path-matches (fn [path] (or
+                                   (get locked-paths path)
+                                   (some #(string/starts-with? (str % "/") path) locked-paths) ;; locked path + '/' prefix of new path
+                                   (some #(string/starts-with? (str path "/") %) locked-paths) ;; new path + '/' prefix of locked path
```

Comment:
This is basically the other half of the thing -- the first line of this is unchanged, just reformatted, but the second one checks the opposite direction of the prefix.